core/Plugin.php

```diff
@@ -461,6 +463,38 @@ public function getMissingDependenciesAsString($piwikVersion = null)
         ));
     }
 
+    /**
+     * Schedules re-archiving of this plugin's reports from when this plugin was last
+     * deactivated to now. If the last time core:archive was run is earlier than the
+     * plugin's last deactivation time, then we use that time instead.
+     *
+     * Note: this only works for CLI archiving setups.
+     *
+     * Note: the time frame is limited by the `[General] rearchive_reports_in_past_last_n_months`
+     * INI config value.
+     *
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    public function schedulePluginReArchiving()
+    {
+        $lastDeactivationTime = $this->getPluginLastDeactivationTime();
+
+        $dateTime = null;
+
+        $lastCronArchiveTime = (int) Option::get(CronArchive::OPTION_ARCHIVING_FINISHED_TS);
+        if (empty($lastCronArchiveTime)) {
+            $dateTime = $lastDeactivationTime;
+        } else if (empty($lastDeactivationTime)) {
+            $dateTime = Date::factory($lastCronArchiveTime);
```

Comment:
@diosmosis if `Option::get(CronArchive::OPTION_ARCHIVING_FINISHED_TS); ` returns `false` then here it would end up being `Date::factory(0)` which may cause issues? It would be 

```
The date '1970-01-01' is a date before first website was online. Try date that's after Aug 6, 1991 (timestamp 681436800).: 0
```

Reply:
in that case the first if branch should be hit (`empty($lastCronArchiveTime)`)

Reply:
👍  didn't check but might be good to have a test for it if it's not the case yet

Reply:
should be covered by the first test case in PluginTest

Reply:
I double checked and it does 
![image](https://user-images.githubusercontent.com/273120/99453651-23eea480-298a-11eb-821b-bcec0abe6df0.png)